node-red-contrib-file/node-log-line.js

```diff
@@ -23,9 +24,49 @@ module.exports = function(RED) {
     RED.nodes.registerType("log-line", register, {})
 }
 
-encapsulate = function(strToEncapsulate) {
-    if (!strToEncapsulate) {
-      return "";
+buildLogstr = function(shouldEncapsulate, config, logstr, separate, data, nb) {
+    var encapsulateStr = '';
+    if (shouldEncapsulate === true) {
+        encapsulateStr = '"';
+    }
+
+    for (let temp of config.template) {
+        switch(temp.typed) {
+          case 'date':
+              logstr += encapsulateStr + encapsulate(shouldEncapsulate, String(Date.now())) + encapsulateStr + separate;
+              break;
+          case 'option_date':
+              logstr += encapsulateStr + encapsulate(shouldEncapsulate, (new Date()).toISOString()) + encapsulateStr + separate;
+              break;
+          case 'option_carr':
+              logstr += encapsulateStr + encapsulate(shouldEncapsulate, data.user.address.carrier) + encapsulateStr + separate;
+              break;
+          case 'option_conv':
+              logstr += encapsulateStr + encapsulate(shouldEncapsulate, data.user.address.conversation.id) + encapsulateStr + separate;
+              break;
+          case 'option_userid':
+              logstr += encapsulateStr + encapsulate(shouldEncapsulate, data.user.id) + encapsulateStr + separate;
+              break;
+          case 'option_userna':
+              logstr += encapsulateStr + encapsulate(shouldEncapsulate, data.user.name) + encapsulateStr + separate;
+              break;
+          case 'str':
+              let cont = config.content[nb];
+              logstr += encapsulateStr;
+              logstr += encapsulate(shouldEncapsulate, (cont.typed === 'msg') ? helper.getByString(data, cont.value) : cont.value);
+              logstr += encapsulateStr + separate ;
+              nb++;
+        }
+    }
+
+    return logstr;
+}
+
+encapsulate = function(shouldEncapsulate, strToEncapsulate) {
```

Comment:
`encapsulate` function should take the boolean `shouldEncapsulate` as a parameter; so it could directly process the string operation.

Reply:
PR has been update because encapsulate function was badly named.